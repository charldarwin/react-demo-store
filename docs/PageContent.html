<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: PageContent</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: PageContent</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>PageContent<span class="signature">()</span><span class="type-signature"></span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="PageContent"><span class="type-signature"></span>new PageContent<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>Component for page content.
This component request 3 props</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_pageContent.jsx.html">components/pageContent.jsx</a>, <a href="components_pageContent.jsx.html#line40">line 40</a>
    </li></ul></dd>
    

    

    

    
</dl>



















    <h5>Example</h5>
    
    <pre class="prettyprint"><code>const title="Home";const details="Text content for page";const repeatDetails; // Useful to repeat details content 4 timesreturn (  &lt;PageContent title={title} details={details} repeatDetails />)</code></pre>



    
    </div>

    

    

    

    

    

    

    

    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="AboutusPage.html">AboutusPage</a></li><li><a href="Banner.html">Banner</a></li><li><a href="Footer.html">Footer</a></li><li><a href="Header.html">Header</a></li><li><a href="IndexPage.html">IndexPage</a></li><li><a href="Layout.html">Layout</a></li><li><a href="NavigateBlock.html">NavigateBlock</a></li><li><a href="NotFoundPage.html">NotFoundPage</a></li><li><a href="PageContent.html">PageContent</a></li><li><a href="PageTitle.html">PageTitle</a></li><li><a href="Product.html">Product</a></li><li><a href="ProductAlert.html">ProductAlert</a></li><li><a href="Products.html">Products</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.4</a> on Tue Apr 14 2020 23:12:50 GMT-0300 (GMT-03:00)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>