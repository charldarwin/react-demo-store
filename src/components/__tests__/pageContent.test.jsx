import React from "react"
import renderer from "react-test-renderer"
import PageContent from "../pageContent"

describe("PageContent component", () => {
    it("Renders", () => {
        const tree = renderer
            .create(<PageContent title="Simple page" details="Lorem content for simple sample page" />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})