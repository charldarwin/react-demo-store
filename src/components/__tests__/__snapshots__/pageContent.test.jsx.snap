// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContent component Renders 1`] = `
<section
  className="sc-AxiKw dNQpti"
>
  <div
    className="sc-AxjAm becqpx"
  >
    <h2
      className="sc-AxirZ ewkXco"
    >
      Simple page
    </h2>
    <div
      className="sc-AxhCb hCbHFA"
    >
      Lorem content for simple sample page
    </div>
  </div>
</section>
`;
